xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/uorgsuggest.vm

```diff
@@ -44,9 +44,19 @@
   ## query.bindValue('input').anyChars().literal($input).anyChars().query()
   ## as soon as we fix XWIKI-15362 (An XWQL query that selects from an object property fails when using the advanced
   ## query parameter API)
-  #set ($query = $query.setLimit(10).bindValue('input', "%$input%"))
+  #set ($query = $query.bindValue('input', "%$input%"))
   #set ($rawResults = $query.execute())
-  #displayResults($rawResults $wikiReference)
+  #set ($results = [])
+  ## Iterate over the results and select the first 10 results that are accessible to the current user.
+  #foreach($rawResult in $rawResults)
+    #if($results.size() >= 10)
+      #break
+    #end
+    #if ($services.security.authorization.hasAccess('view', $rawResult))
+      #set ($discard = $results.add($rawResult))
+    #end
+  #end
+  #displayResults($results $wikiReference)
```

Comment:
Tested with 200k users and removing the limit in the query does not have a measurable impact on the script response time.